gateway/src/main/java/io/camunda/zeebe/gateway/impl/probes/health/ClusterHealthIndicator.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.gateway.impl.probes.health;
+
+import static java.util.Objects.requireNonNull;
+
+import io.camunda.zeebe.gateway.impl.broker.cluster.BrokerClusterState;
+import io.camunda.zeebe.protocol.record.PartitionHealthStatus;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+
+/**
+ * The cluster health indicator signals if there are still any healthy partition available, if not
+ * then is set as down as no processing is happening. In the details also indicates the health
+ * status of all partitions.
+ */
+public class ClusterHealthIndicator implements HealthIndicator {
+
+  private final Supplier<Optional<BrokerClusterState>> clusterStateSupplier;
+  private boolean processing = false;
+
+  public ClusterHealthIndicator(final Supplier<Optional<BrokerClusterState>> clusterStateSupplier) {
+    this.clusterStateSupplier = requireNonNull(clusterStateSupplier);
+  }
+
+  @Override
+  public Health health() {
+    final var optClusterState = clusterStateSupplier.get();
+
+    if (optClusterState.isEmpty()) {
+      return Health.down().build();
+    } else {
+      if (optClusterState.get().getBrokers().isEmpty()) {
+        return Health.down().build();
+      } else {
+        if (!optClusterState.get().getPartitions().isEmpty()) {
+          processing = false;
+          final List<Integer> partitions = optClusterState.get().getPartitions();
+
+          final Map<String, PartitionHealthStatus> partitionDetails =
+              getPartitionsHealthStatus(partitions, optClusterState);
+
+          if (!processing) {
+            return Health.down().withDetails(partitionDetails).build();
+          } else {
+            return Health.up().withDetails(partitionDetails).build();
+          }
```

Comment:
:x: I recommend to return a `DEGRADED` state. 
```suggestion
private Health getStatus(
      final int partitionCount, final Map<String, PartitionHealthStatus> partitionDetails) {

    final var unhealthyPartitionsCount =
        partitionDetails.values().stream()
            .filter(p -> p == PartitionHealthStatus.UNHEALTHY || p == PartitionHealthStatus.DEAD)
            .count();

    final var healthyPartitionsCount =
        partitionDetails.values().stream().filter(p -> p == PartitionHealthStatus.HEALTHY).count();

    final var hasOnlyUnhealthyPartitions = unhealthyPartitionsCount == partitionDetails.size();
    final var hasOnlyHealthyPartitions = healthyPartitionsCount == partitionCount;

    if (hasOnlyUnhealthyPartitions) {
      return Health.down().withDetails(partitionDetails).build();
    } else if (hasOnlyHealthyPartitions) {
      return Health.up().withDetails(partitionDetails).build();
    } else {
      return Health.status("DEGRADED").withDetails(partitionDetails).build();
    }
  }
```

This means, we are defining a new "Status" code. I think this is ok. Please read about it [here](https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html#actuator.endpoints.health.writing-custom-health-indicators) and verify how it affects the overall health indicator.

Reply:
In additions, if there are missing partitions, it should return `DEGRADED` even if all other partitions are healthy.

Reply:
I was looking at the docs that you linked and they suggested adding a custom StatusAgregator or a yaml config to agregate/order the new status. In the sense that it is "required in order to combine subsystem states expressed through getStatus() into one state for the entire system" according to the docs.

Since we don't use this status as a subcomponent of another health status, then we don't need this aggregator correct?